crates/ironrdp-rdpdr/src/pdu/efs.rs

```diff
@@ -2391,3 +2440,433 @@ impl ClientDriveQueryDirectoryResponse {
         }
     }
 }
+
+/// [2.2.3.3.6] Server Drive Query Volume Information Request
+///
+/// We only need to read the buffer up to the FileInformationClass to get the job done, so the rest of the fields in
+/// this structure are discarded. See FreeRDP:
+/// https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L464
+///
+/// [2.2.3.3.6]: https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-rdpefs/484e622d-0e2b-423c-8461-7de38878effb
+#[derive(Debug)]
+pub struct ServerDriveQueryVolumeInformationRequest {
+    pub device_io_request: DeviceIoRequest,
+    pub fs_info_class_lvl: FileSystemInformationClassLevel,
+}
+
+impl ServerDriveQueryVolumeInformationRequest {
+    const NAME: &str = "DR_DRIVE_QUERY_VOLUME_INFORMATION_REQ";
+    const FIXED_PART_SIZE: usize = 4 /* FsInformationClass */ + 4 /* Length */ + 24 /* Padding */;
+
+    pub fn decode(dev_io_req: DeviceIoRequest, src: &mut ReadCursor<'_>) -> PduResult<Self> {
+        ensure_size!(in: src, size: Self::FIXED_PART_SIZE);
```

Comment:
Nitpick: we have `ensure_fixed_part_size!(in: src)`, specifically for this. I’ll not block the PR for that, feel free to send a follow up PR if you prefer.

Reply:
I just noticed that, and patched all those up in https://github.com/Devolutions/IronRDP/pull/298